### ISSUE TITLE
If phpMussel can detected  Macro virus in Microsoft Document ? 

### ISSUE THREAD

**Issue by 1nFrastr** [phpMussel/phpMussel#165]
## Server information

phpMussel v1.5.0

php 5.6

WAMP SERVER on win7

## Issue

If phpMussel can detected  Macro virus in Microsoft Document ? 

  **Wating for your reply , thanks !**

**Comment by Maikuolan** [comment]
Yes and no. phpMussel is capable is seeing inside Microsoft Documents, to be able to scan for malicious macros, but the process needs to be improved, I think. Currently, phpMussel is capable of detecting most of the more severe macro viruses currently in circulation, but also, it isn't 100% comprehensive (i.e., it's possible that some less common things might not be detected sometimes). Also, it isn't so good at knowing the difference between a benign/safe macro, and other unrelated objects in the document (so, it can detect malicious macros easily, due to the way it processes Microsoft documents and other similar documents that support macros, but it isn't so easy yet for users to say "I want to block any documents containing any macros, but allow any documents not containing macros").

In short: It works, but I'm not 100% satisfied with it yet. When there's more time (or earlier, if someone with more experience that could help code it comes along and is willing to help), hopefully we can improve it, or rewrite it in a better way.

(There's also some more discussion about this in #106).

**Comment by pkcnguyen** [comment]
Hi Maikuolan,

I am trying to run phpMussel on XAMPP.  According to your directions on https://github.com/phpMussel/phpMussel/tree/master/vault and https://github.com/phpMussel/phpMussel/blob/master/_docs/readme.en.md#SECTION2 , we rename the config.ini.RenameMe to config.ini.  Can you please let us know where to insert the config.ini file into the XAMPP folder on Windows?  Also, do we put all of your files in 1 directory or name the subdirectories as you posted on https://github.com/phpMussel/phpMussel/tree/master/vault ?

Waiting for your reply.  Thank you very much for your help!

**Comment by Maikuolan** [comment]
Hi @pkcnguyen.

Sorry for the delayed reply. Busy day today.

After renaming the file, it stays in the vault directory, alongside everything else already in the vault directory. Other than renaming the configuration file (as per the documentation instructions), the package structure doesn't need to be modified at all in order to work. :-)

Of course, the documentation is for humans, so, you won't need to upload any of that (i.e., the "_docs" directory and such is for you, so your server won't need it to make things works). In most cases, just the "loader.php" file and the "vault" directory (as is, structurally) will be all that's needed.

phpMussel neither requires nor is dependent upon any specific server stack, and should work equally well, whether it be for WAMP, MAMP, XAMP, XAMPP, Nginx, Apache, or whatever else. In most cases, the process for getting it installed and working correctly should be the same across all stacks (of course, there may be exceptions for some specific stacks that work really differently than normal, and there are probably stacks I haven't seen nor experimented with before).

Let me know if there are any problems at all.

(Probably better to create a new issue next time though, instead of replying to an issue which is unrelated to the problem you need help with, to keep everything here a little more organised, with cleaner issue issues and such, and also to prevent potentially annoying other users; when someone creates a new issue, or replies to an issue, they automatically subscribe to it, which means that they'll receive email notifications for any future replies written to the issue; this means that anyone subscribed to an issue, will receive email notifications unrelated to what they originally posted, due to the unrelated replies to their issue. Just creating a new issue instead should keep all the notifications separate though, and help to prevent annoying anyone).

**Comment by pkcnguyen** [comment]
Hi Maikuolan,

No worries at all.  Sorry for posting in another thread.  It did not allow me to create a separate thread unless I create new code or project or something like that.  I appreciate the response.  I will go ahead and try it by moving the vault directory into local file in XAMPP.  If I run into other problems, I will let you know.

Thank you again,

Peter

**Comment by Maikuolan** [comment]
No problem, and no worries. Fingers crossed that all goes well. :-)

**Comment by pkcnguyen** [comment]
Thank you again for your fingers crossed, Maikuolan!  Trying to put everything together right now.  If I run into anymore trouble I'll let you know again. Fingers crossed.  Thanks a mil!  :)

**Comment by Maikuolan** [comment]
@1nFrastr I've just committed some changes tonight which should help improve phpMussel's ability to properly detect macros. :-)

There's also a new configuration directive added to allow users to block any files containing macros. I've marked it as "experimental" for now, due to that it is very new, and ideally I would like to do more testing to confirm that it actually blocks all of the most common types of macros found in documents and spreadsheets before removing the "experimental" marker from it, but in all of my local testing thus far, it looks promising, and seems to work as expected.

I'll mark this issue as "resolved", seeing as this feature is technically now supported, and close this issue. If you encounter any problems with it though, feel free to open a new issue. Possibly also there'll be improvements in the near future based on the results of further testing and on whatever else can be discovered.
